### ISSUE TITLE
Extreme lag, stuttering, freezes, slow animations on Wayland.

### ISSUE THREAD

**Issue by iceeburr** [kpcyrd/spotify-launcher#38]
From what the title says, all of the UI is slow, doesn't respond, some is even broken and you can't click on it. The whole app lags like crazy and it's to the point where even switching the song is impossible, don't even want to talk about extensions from spicetify, those might even take 20 seconds to load from start point.
Have tested on this app and also the AUR `spotify` app which has the same problem. I hope this can be fixed in the future. I am opening this issue because I couldn't find anything else about this type of behavior online.
To be more specific I am running Hyprland `hyprland-git` on Arch Linux.
I have passed the needed flags to the app in the configuration:
```conf
[spotify]
extra_arguments = ["--enable-features=UseOzonePlatform", "--ozone-platform=wayland"]
```
Without enabling those the application appears as "pixelated" and "low quality", which is common for electron apps on xwayland. I actually notice a very similar pattern for other electron apps, they also start to lag like your little brother playing fortnite in the other room, for Discord it's not that bad, but bro for Spotify it's literally impossible to use. It also uses a lot of RAM which is just impossible. Like why do I even need the spotify desktop client anymore when it's unusable.. If you really need a video of this tell me I'll post one, but I'm 99% sure you will be able to replicate it.

**Comment by iceeburr** [comment]
I found the reason it happens, sadly not a solution yet.
I have a Hybrid laptop. I have a RTX 3070 ti with an AMD GPU. I use the supergfxctl utility to switch between those modes. Whenever I am in Hybrid mode it loads both drivers. That's when funky stuff happen. Even if the GPU is not being used and Hyprland was specifically configured to use the AMD GPU it will create this lag. Switch to Integrated mode (or VFIO) and it works like a charm. I will look for a solution if I even find anything. If not this is the only way then.
This affects all applications and is not limited to spotify. It is mostly noticeable in electron apps.

**Comment by iceeburr** [comment]
I was able to fix the issue, not the best solution, but hey "it works on my machine". The whole issue is probably because the app uses software rendering. It happens randomly, I don't know why. How I fixed it is simple. Don' try to run it in Wayland mode. It will lag. I tried using switcherooctl (utility used to force applications to use a specific GPU). When in xwayland it is able to use the dGPU, it also doesn't lag and works as expected. The whole issue with xwayland is that it has scaling issues and applications look pixelated. Happily I am a Hyprland user and we have a neat little setting to counter that. Simple add these to your Hyprland config and you are good to go 👍🏻
```
xwayland {
  force_zero_scaling = true
}
env = GDK_SCALE,2
env = XCURSOR_SIZE,24
```
Change GDK scale to whatever u need. 2 works fine for me. Remember u can press CTRL and +/- in apps to zoom in/out. Doesn't work on all, but for most it does. If numpad doesn't work try the keys near backspace, if that doesn't work try pressing the ALT key, if a menu on the top appears try to find a "zoom" option and use that. If that doesn't work then u can give up because the app doesn't support zooming in.
On spotify this "zoom" works.
